test/test_custom_ops.py

```diff
@@ -2137,6 +2137,61 @@ def _(x, y):
         self.assertEqual(z, x + y)
         self.assertTrue(cpu_called)
 
+    def test_supports_tensorlist(self):
+        @torch._library.autograd.supports_tensorlist
+        class Stack(torch.autograd.Function):
+            @staticmethod
+            def forward(ctx, xs):
+                return torch.stack(xs)
+
+            @staticmethod
+            def backward(ctx, grad):
+                return list(grad.unbind(0))
+
+        # call two applys, do a backward on the first
+        def t():
+            return torch.randn([], requires_grad=True)
+
+        xs0 = [t(), t(), t()]
+        xs1 = [t(), t(), t(), t()]
+        y0 = Stack.apply(xs0)
+        y1 = Stack.apply(xs1)
+        grads = torch.autograd.grad(y0.sum(), xs0)
+        self.assertEqual(grads, [torch.tensor(1.0) for _ in range(3)])
+
+        # call one apply, do multiple backwards
+        xs = [t(), t(), t()]
+        y = Stack.apply(xs)
+        _ = torch.autograd.grad(y.sum(), xs, retain_graph=True)
+        _ = torch.autograd.grad(y.sum(), xs, retain_graph=True)
+        grads = torch.autograd.grad(y.sum(), xs, retain_graph=True)
+        self.assertEqual(grads, [torch.tensor(1.0) for _ in range(3)])
+
+        # error: on access forward, backward directly
+        with self.assertRaisesRegex(NotImplementedError, "Function.forward directly"):
+            Stack.forward(None, xs)
+        with self.assertRaisesRegex(NotImplementedError, "Function.backward directly"):
+            Stack.backward(None, xs)
+
+        # the recursive case
+        @torch._library.autograd.supports_tensorlist
+        class Foo(torch.autograd.Function):
+            @staticmethod
+            def forward(ctx, xs):
+                if len(xs) > 0:
+                    return Foo.apply(xs[1:])
```

Comment:
Is doing a Foo.apply() during the "backward()" call ok? I would expect it to be a more likely thing users will do than this one.

Reply:
Yes, I can add a test for this